text/3-entrees/storage-api/storage-maps.md

```diff
@@ -0,0 +1,58 @@
+# Storage Maps
+*[`pallets/simple-map`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/simple-map)*
+
+In the appetizer on [storage values](../../2-appetizers/2-storage-values.md) we learned to store a single value in blockchain storage to be persisted between blocks. In this recipe, we will see how to store a mapping from keys to values, similar to Rust's own [`HashMap`](https://doc.rust-lang.org/std/collections/struct.HashMap.html).
+
+## Declaring a `StorageMap`
+
+We declare a single storage map with the following syntax.
+
+```rust, ignore
+decl_storage! {
+	trait Store for Module<T: Trait> as SimpleMap {
+		SimpleMap get(fn simple_map): map hasher(blake2_128_concat) T::AccountId => u32;
+	}
+}
+```
+
+Much of this should look familiar to you from storage values. Reading the line from left to right we have:
+* `StorageMap` - the name of the storage map
+* `get(fn simple_map)` - the name of a getter function that will return values from the map.
+* `: map hasher(blake2_128_concat)` - beginning of the type declaration. This is a map and it will use the [`blake2_128_concat`](https://substrate.dev/rustdocs/master/frame_support/trait.Hashable.html#tymethod.blake2_128_concat) hasher. More on this below.
+* `T::AccountId => u32` - The specific key and value tyes of the map. This is a map from `AccountId`s to `u32`s.
+
+## Choosing a Hasher
+
+Although the syntax above is complex, most of it should be straightforward if you've understood the recipe on storage values. The last unfamiliar piece of writing a storage map is choosing which hasher to use. In general you should choose one of the three following hashers. The coice of hasher will affect the performance and security of your chain. If you don't want to think much about this, just choose `blake2_128_concat` and skip to the next section.
+
+### `blake2_128_concat`
+
+This is a cryptographically secure hash function, and is always safe to use. If is reasonably efficient, and will keep your storage tree balanced. You _must_ choose this hasher if users of your chain have the ability to affect the storage keys. In this pallet, the keys are `AccountId`s. At first it may _seem_ that the user doesn't affect the `AccountId`, but in reality a malicious user can generate thousands of accounts and use the one that will affect the chain's storage tree in the way the attacker likes. For this reason, we have chosen to use the `blake2_128_concat` hasher.
```

Comment:
```suggestion
This is a cryptographically secure hash function, and is always safe to use. It is reasonably efficient, and will keep your storage tree balanced. You _must_ choose this hasher if users of your chain have the ability to affect the storage keys. In this pallet, the keys are `AccountId`s. At first it may _seem_ that the user doesn't affect the `AccountId`, but in reality a malicious user can generate thousands of accounts and use the one that will affect the chain's storage tree in the way the attacker likes. For this reason, we have chosen to use the `blake2_128_concat` hasher.
```